### PR TITLE
Be explicit about errors

### PR DESCRIPTION
**Pull request by bissenbay**
Related-To: https://github.com/thoth-station/package-analyzer/issues/114

### CODE REVIEW

File: thoth/python/artifact.py [hunk 1]
```diff
@@ -71,15 +71,19 @@ def _extract_py_module(self) -> None:
...
+                    with zipfile.ZipFile(self.compressed_file) as zip_ref:
+                        zip_ref.extractall(self.dir_name)
+                        _LOGGER.debug("Artifact is .%r file", ext)
             except Exception as e:
-                tf = tarfile.open(self.compressed_file)
-                tf.extractall(self.dir_name)
-                _LOGGER.debug("Artifact is .tar.gz")
-        except Exception as e:
-            _LOGGER.error("Could not create temp dir")
+                _LOGGER.exception(f"Could not extract {self.compressed_file}: {str(e)}")
```
Comment (bissenbay) [inline thoth/python/artifact.py:84]: @fridex I put `_LOGGER.exception` here. Then, the log was below:
`
(package-analyzer) [bdauletb@bdauletb package-analyzer]$ ./thoth-package-analyzer python --package-name pygobject --package-version 2.28.3 > out.json
2020-04-20 19:12:27,297 12308 WARNING  thoth.common:317: Logging to a Sentry instance is turned off
2020-04-20 19:12:27,297 12308 INFO     thoth.common:339: Logging to rsyslog endpoint is turned off
2020-04-20 19:12:27,298 12308 INFO     __main__:69: Version: 0.1.0
2020-04-20 19:12:27,298 12308 INFO     thoth.package_analyzer.python.fetch:39: Fetching digests for package 'pygobject' in version '2.28.3' from 'https://pypi.org/simple'
2020-04-20 19:12:27,938 12308 ERROR    thoth.python.artifact:84: Could not extract /tmp/tmpsxqkqy1p: File is not a zip file
Traceback (most recent call last):
  File "/home/bdauletb/.local/share/virtualenvs/package-analyzer-CT5UWDp5/lib/python3.6/site-packages/thoth/python/artifact.py", line 80, in _extract_py_module
    with zipfile.ZipFile(self.compressed_file) as zip_ref:
  File "/usr/lib64/python3.6/zipfile.py", line 1131, in __init__
    self._RealGetContents()
  File "/usr/lib64/python3.6/zipfile.py", line 1198, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
2020-04-20 19:12:28,465 12308 ERROR    thoth.python.artifact:84: Could not extract /tmp/tmp1ggsdb_0: File is not a zip file
Traceback (most recent call last):
  File "/home/bdauletb/.local/share/virtualenvs/package-analyzer-CT5UWDp5/lib/python3.6/site-packages/thoth/python/artifact.py", line 80, in _extract_py_module
    with zipfile.ZipFile(self.compressed_file) as zip_ref:
  File "/usr/lib64/python3.6/zipfile.py", line 1131, in __init__
    self._RealGetContents()
  File "/usr/lib64/python3.6/zipfile.py", line 1198, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
`
And also I saw `out.json` file was generated as well. The only is that `digests` was empty which is expected. Result:
`json
{
  "metadata": {
    "analyzer": "thoth-package-analyzer",
    "analyzer_version": "0.1.0",
    "arguments": {
      "python": {
        "dry_run": false,
        "index_url": "https://pypi.org/simple",
        "no_pretty": false,
        "output": null,
        "package_name": "pygobject",
        "package_version": "2.28.3"
      },
      "thoth-package-analyzer": {
        "verbose": false
      }
    },
    "datetime": "2020-04-20T19:12:28.465658",
    "distribution": {
      "codename": "Ootpa",
      "id": "rhel",
      "like": "fedora",
      "version": "8.0",
      "version_parts": {
        "build_number": "",
        "major": "8",
        "minor": "0"
      }
    },
    "document_id": null,
    "duration": 1,
    "hostname": "bdauletb.remote.csb",
    "os_release": {
      "id": "rhel",
      "name": "Red Hat Enterprise Linux",
      "platform_id": "platform:el8",
      "redhat_bugzilla_product": "Red Hat Enterprise Linux 8",
      "redhat_bugzilla_product_version": "8.0",
      "redhat_support_product": "Red Hat Enterprise Linux",
      "redhat_support_product_version": "8.0",
      "version": "8.0 (Ootpa)",
      "version_id": "8.0"
    },
    "python": {
      "api_version": 1013,
      "implementation_name": "cpython",
      "major": 3,
      "micro": 8,
      "minor": 6,
      "releaselevel": "final",
      "serial": 0
    },
    "thoth_deployment_name": null,
    "timestamp": 1587409948
  },
  "result": {
    "artifacts": [
      {
        "digests": [],
        "name": "pygobject-2.28.3.win32-py2.6.msi",
        "sha256": "0ad6f157cbbc292bbdd7e979b85dbfe79d93fa3f959d3140cb430c786750b13f",
        "symbols": {}
      },
      {
        "digests": [],
        "name": "pygobject-2.28.3.win32-py2.7.msi",
        "sha256": "6ec3bd44ee3b5fbb26b29e1cda50bef863fd775225850a2b7a94040ffd653e1a",
        "symbols": {}
      }
    ],
    "error": false,
    "error_message": "",
    "index_url": "https://pypi.org/simple",
    "package_name": "pygobject",
    "package_version": "2.28.3"
  }
}
`

Comment (bissenbay) [reply thoth/python/artifact.py:84]: And also I saw `out.json` file was generated as well. The only is that `digests` was empty which is expected. Result:
```json
{
  "metadata": {
    "analyzer": "thoth-package-analyzer",
    "analyzer_version": "0.1.0",
    "arguments": {
      "python": {
        "dry_run": false,
        "index_url": "https://pypi.org/simple",
        "no_pretty": false,
        "output": null,
        "package_name": "pygobject",
        "package_version": "2.28.3"
      },
      "thoth-package-analyzer": {
        "verbose": false
      }
    },
    "datetime": "2020-04-20T19:12:28.465658",
    "distribution": {
      "codename": "Ootpa",
      "id": "rhel",
      "like": "fedora",
      "version": "8.0",
      "version_parts": {
        "build_number": "",
        "major": "8",
        "minor": "0"
      }
    },
    "document_id": null,
    "duration": 1,
    "hostname": "bdauletb.remote.csb",
    "os_release": {
      "id": "rhel",
      "name": "Red Hat Enterprise Linux",
      "platform_id": "platform:el8",
      "redhat_bugzilla_product": "Red Hat Enterprise Linux 8",
      "redhat_bugzilla_product_version": "8.0",
      "redhat_support_product": "Red Hat Enterprise Linux",
      "redhat_support_product_version": "8.0",
      "version": "8.0 (Ootpa)",
      "version_id": "8.0"
    },
    "python": {
      "api_version": 1013,
      "implementation_name": "cpython",
      "major": 3,
      "micro": 8,
      "minor": 6,
      "releaselevel": "final",
      "serial": 0
    },
    "thoth_deployment_name": null,
    "timestamp": 1587409948
  },
  "result": {
    "artifacts": [
      {
        "digests": [],
        "name": "pygobject-2.28.3.win32-py2.6.msi",
        "sha256": "0ad6f157cbbc292bbdd7e979b85dbfe79d93fa3f959d3140cb430c786750b13f",
        "symbols": {}
      },
      {
        "digests": [],
        "name": "pygobject-2.28.3.win32-py2.7.msi",
        "sha256": "6ec3bd44ee3b5fbb26b29e1cda50bef863fd775225850a2b7a94040ffd653e1a",
        "symbols": {}
      }
    ],
    "error": false,
    "error_message": "",
    "index_url": "https://pypi.org/simple",
    "package_name": "pygobject",
    "package_version": "2.28.3"
  }
}
`

Comment (fridex) [reply thoth/python/artifact.py:84]: Sounds good :+1: 

Is it possible to extract also the file it tried to extract? Or is the file corrupted in some way?

File: thoth/python/artifact.py [hunk 2]
```diff
@@ -71,15 +71,19 @@ def _extract_py_module(self) -> None:
...
+        except Exception as exc:
+            _LOGGER.warning(f"Could not create temp dir: {str(exc)}")
```
Comment (bissenbay) [inline thoth/python/artifact.py:86]: @fridex Should I use `_LOGGER.exception` here as well?

Comment (fridex) [reply thoth/python/artifact.py:86]: Yes, please do so. The exception will be reported to sentry and we will have the whole traceback of the error that happened.

File: thoth/python/artifact.py (hunk 1 above)
Comment (bissenbay) [reply thoth/python/artifact.py:84]: I'm not sure if I got the question @fridex . We're using `zipfile` for extracting files with `.whl`, `.zip`, `.egg`, `.exe` extensions and `tarfile` for extracting only files with `.tar.gz`.
For other extensions, like `.msi`, `rpm`,  both `zipfile` and `tarfile` don't work, they fail with error message noting that this is a not zip file.
So, do you want me to dig into if there any other libs to use for extracting `.msi`, `.rpm` packages?

File: thoth/python/artifact.py (hunk 2 above)
Comment (bissenbay) [reply thoth/python/artifact.py]: > Yes, please do so. The exception will be reported to sentry and we will have the whole traceback of the error that happened.

Done!

File: thoth/python/artifact.py (hunk 1 above)
Comment (fridex) [reply thoth/python/artifact.py:84]: OK, I see your point now. There are `msi` files that are hosted:

https://pypi.org/simple/pygobject/

What about skipping these files and reporting error? It feels to me that they are leftovers from old PyPI (but I'm not 100% sure here).

Comment (bissenbay) [reply thoth/python/artifact.py:84]: Yes, @fridex , these two packages `pygobject-2.28.3.win32-py2.6.msi` and 
`pygobject-2.28.3.win32-py2.7.msi` is the reason for triggering [this issue](https://github.com/thoth-station/package-analyzer/issues/114).

File: thoth/python/artifact.py [hunk 3]
```diff
@@ -71,15 +71,22 @@ def _extract_py_module(self) -> None:
...
-                with zipfile.ZipFile(self.compressed_file) as zip_ref:
-                    zip_ref.extractall(self.dir_name)
-                    _LOGGER.debug("Artifact is .whl")
+                if self.compressed_file.endswith('.tar.gz'):
+                    tf = tarfile.open(self.compressed_file)
+                    tf.extractall(self.dir_name)
+                    _LOGGER.debug("Artifact is .tar.gz file")
+                else:
+                    ext = self.compressed_file.split('.')[-1]
+                    if ext in ['.msi', 'rpm']:
```
Comment (bissenbay) [inline thoth/python/artifact.py:80]: @fridex looks like you already approved, :smile: ! But, I wanted to add some logic like this to skip those files.

Comment (bissenbay) [reply thoth/python/artifact.py:80]: Oh let me just revert back, deleting last commit

**Pull request merged**